src/ast.rs

```diff
@@ -0,0 +1,1617 @@
+                    get_relative_path(node._base._arena_id.unwrap(), parent, path_map)
+                }),
+                start: node._base.start,
+                end: node._base.end,
+                raw: node._base.raw.to_owned(),
+                ref_: node.ref_.clone(),
+                resolved: get_relative_path(
+                    node._base._arena_id.unwrap(),
+                    node.resolved.unwrap(),
+                    path_map,
+                ),
+            }))
+        }
+        Node::Character(node) => NodeUnresolved::Character(Box::new(CharacterUnresolved {
+            parent: node
+                ._base
+                .parent
+                .map(|parent| get_relative_path(node._base._arena_id.unwrap(), parent, path_map)),
+            start: node._base.start,
+            end: node._base.end,
+            raw: node._base.raw.to_owned(),
+            value: node.value,
+        })),
+        Node::CharacterSet(node) => {
+            NodeUnresolved::CharacterSet(Box::new(CharacterSetUnresolved {
+                parent: node._base.parent.map(|parent| {
+                    get_relative_path(node._base._arena_id.unwrap(), parent, path_map)
+                }),
+                start: node._base.start,
+                end: node._base.end,
+                raw: node._base.raw.to_owned(),
+                kind: node.kind,
+                strings: node.strings,
+                key: node.key.clone(),
+                value: node.value.clone(),
+                negate: node.negate,
+            }))
+        }
+        Node::Flags(node) => NodeUnresolved::Flags(Box::new(FlagsUnresolved {
+            parent: node
+                ._base
+                .parent
+                .map(|parent| get_relative_path(node._base._arena_id.unwrap(), parent, path_map)),
+            start: node._base.start,
+            end: node._base.end,
+            raw: node._base.raw.to_owned(),
+            dot_all: node.dot_all,
+            global: node.global,
+            has_indices: node.has_indices,
+            ignore_case: node.ignore_case,
+            multiline: node.multiline,
+            sticky: node.sticky,
+            unicode: node.unicode,
+            unicode_sets: node.unicode_sets,
+        })),
+    }
+}
+
+#[derive(Clone)]
+pub struct RegExpLiteral {
+    _base: NodeBase,
+    pub pattern: Id<Node>, /*Pattern*/
+    pub flags: Id<Node>,   /*Flags*/
+}
+
+#[cfg(test)]
+fn deserialize_possibly_infinity_u32<'de, D>(deserializer: D) -> Result<u32, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    #[derive(Deserialize)]
+    #[serde(untagged)]
+    enum StringOrU32 {
+        String(String),
+        U32(u32),
+    }
+    let string_or_u32 = StringOrU32::deserialize(deserializer)?;
+    Ok(match string_or_u32 {
+        StringOrU32::String(value) => {
+            // TODO: should handle this better?
+            assert!(value == "$$Infinity");
+            u32::MAX
```

Comment:
I don't know if I actually "tested" this anywhere, basically I just tried to use `u32::MAX` in places where the JS version was using `Infinity`